### PR TITLE
feat: add devcontainer-image flag to override container image in workspaces

### LINKED ISSUE #299: feature: custom image for docker provider
Add a configuration in the docker provider to use a custom image instead of the one that is inferred or choosen by devcontainer.json

This could be useful in case the user uses personalized development images that wants to keep consistent between multiple projects without necessarly modifying the devcontainer.json in a repo

### PR DESCRIPTION
**Pull request by 89luca89**
This PR adds support for a `--devcontainer-image` flag in the `up` command that will let the user override the container image with whatever image they want

Fix #299 
Resolves ENG-1417

### CODE REVIEW

File: e2e/tests/up/up.go [hunk 1]
```diff
@@ -503,6 +503,34 @@ var _ = DevPodDescribe("devpod up test suite", func() {
...
+				tempDir, err := framework.CopyToTempDir("tests/up/testdata/docker")
+				framework.ExpectNoError(err)
+				ginkgo.DeferCleanup(framework.CleanupTempDir, initialDir, tempDir)
+
+				f := framework.NewDefaultFramework(initialDir + "/bin")
+
+				_ = f.DevPodProviderDelete(ctx, "docker")
+				err = f.DevPodProviderAdd(ctx, "docker")
+				framework.ExpectNoError(err)
+				err = f.DevPodProviderUse(context.Background(), "docker")
```
Comment (ThomasK33) [inline e2e/tests/up/up.go:516]: Shouldn't this also use `ctx`?

```suggestion
				err = f.DevPodProviderUse(ctx, "docker")
```

**Review by ThomasK33 — COMMENTED** [review]
LGTM

**Review by ThomasK33 — APPROVED** [review]

**Review by pbialon — APPROVED** [review]

File: pkg/devcontainer/run.go [hunk 2]
```diff
@@ -106,6 +119,11 @@ func (r *Runner) prepare() (*config.SubstitutedConfig, *WorkspaceConfig, error)
 	if parsedConfig.WorkspaceMount != "" {
 		workspace.WorkspaceMount = parsedConfig.WorkspaceMount
 	}
+
+	if options.DevContainerImage != "" {
+		parsedConfig.ImageContainer = config.ImageContainer{Image: options.DevContainerImage}
```
Comment (FabianKramm) [inline pkg/devcontainer/run.go:124]: What happens when in the devcontainer.json a Dockerfile build is specified? Does this still work then?

Comment (89luca89) [reply pkg/devcontainer/run.go:125]: It was ignored, I've fixed it now :+1:

**Review by FabianKramm — APPROVED** [review]

**Pull request merged**